mountpoint-s3/src/fs.rs

```diff
@@ -366,7 +391,13 @@ where
         // hard links, so we just assume one link for files (itself) and two links for directories
         // (itself + the "." link).
         let (perm, nlink) = match lookup.inode.kind() {
-            InodeKind::File => (self.config.file_mode, 1),
+            InodeKind::File => {
+                if is_flexible_retrieval_storage_class(lookup) {
+                    (0o000, 1)
```

Comment:
If we ever allow over-writing, will this be a breaking change?

Should we just mask `config.file_mode` to disallow rx?